polaris/hub/polarisfs.py

```diff
@@ -120,3 +128,71 @@ def cat_file(
         with fsspec.open(signed_url, "rb", **kwargs) as f:
             data = f.read()
         return data[start:end]
+
+    def rm(self, path: str, recursive: bool = False, maxdepth: Optional[int] = None):
+        """Remove a file or directory from the Polaris dataset.
+
+        This method is provided for compatibility with the Zarr storage interface.
+        It may be called by the Zarr store when removing a file or directory.
+
+        Args:
+            path: The path to the file or directory to be removed.
+            recursive: If True, remove directories and their contents recursively.
+            maxdepth: The maximum depth to recurse when removing directories.
+
+        Returns:
+            None
+
+        Note:
+            This method currently it does not perform any removal operations and is included
+            as a placeholder that aligns with the Zarr interface's expectations.
+        """
+        return
```

Comment:
@cwognum From what I noticed, if you want to write in a directory that already exists, it will try to remove it first. 

For example, if test.zarr directory is found in the dataset and I try to write to that existing path, it calls the `rm` file operation.
<img width="973" alt="Screenshot 2024-03-04 at 3 13 44 PM" src="https://github.com/polaris-hub/polaris/assets/72565412/519a84db-9e99-4dfd-bcd4-9fdaa1106462">

Reply:
Thank you for taking a look! If that's the only case in which it fails, I would prefer to still have it raise an exception to prevent a silent bug later on.